WordPress/gutenberg
# Data views: Update 'Rows per page' view option label to account for Grid layout

## What?
When the grid layout is selected the 'Rows per page' view option now reads 'Entries per page'.

## Why?
In the context of grid layout 'Rows per page' is ambiguous. Does it refer to the individual items or the number of rows in the grid?

## Testing Instructions
1. In the Site editor go to manage pages or templates
2. In view options switch to grid layout
3. Check the 'Entries per page' label

<img width="419" alt="Screenshot 2024-01-30 at 15 37 08" src="https://github.com/WordPress/gutenberg/assets/846565/7ea9570b-1d74-450a-a49a-ec822f6a7a8b">

I'd appreciate input from @WordPress/gutenberg-design on the terminology. Possible alternatives to 'entries':

* 'Records per page'
* 'Items per page'
* $post-type (e.g. 'Patterns per page')
* Something else?

Worth noting that all of these options could effectively account for any layout meaning we could potentially remove the conditional logic and simplify things a bit. I suppose that's a question of consistency versus specificity.
